engine/src/main/scala/cromwell/engine/workflow/tokens/TokenEventLogger.scala

```diff
@@ -7,38 +7,50 @@ import scala.concurrent.duration.FiniteDuration
 
 trait TokenEventLogger {
   def flagTokenHog(hogGroup: String): Unit
+  def getLimitedGroups: Set[String]
+
   def outOfTokens(backend: String): Unit
+  def getLimitedBackends: Set[String]
 }
 
 case object NullTokenEventLogger extends TokenEventLogger {
   override def flagTokenHog(hogGroup: String): Unit = ()
+  override def getLimitedGroups: Set[String] = Set.empty
+
   override def outOfTokens(backend: String): Unit = ()
+  override def getLimitedBackends: Set[String] = Set.empty
 }
 
 class CachingTokenEventLogger(log: LoggingAdapter,
                               cacheEntryTTL: FiniteDuration) extends TokenEventLogger {
 
-  log.info("Using CachingTokenEventLogger with an interval of {} to log token queue events", cacheEntryTTL)
-
-  private val cache = CacheBuilder.newBuilder()
+  private val groupCache = CacheBuilder.newBuilder()
     .expireAfterWrite(cacheEntryTTL._1, cacheEntryTTL._2)
     .maximumSize(10000)
     .build[String, Object]()
 
-
   override def flagTokenHog(hogGroup: String): Unit = {
-    if (Option(cache.getIfPresent("HOG_" + hogGroup)).isEmpty) {
```

Comment:
Removing the `isEmpty` check allows continual, [idempotent](https://javadoc.io/doc/com.google.guava/guava/latest/com/google/common/cache/Cache.html#put(K,V)) `put` operations to constantly keep the `hogGroup` entry fresh in the cache. With the check, there would be a brief gap after the entry expired, but before it was re-added, during which new workflows could inadvertently start.